modules/core/src/main/java/org/apache/ignite/internal/util/StripedExecutorProxy.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.util;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.apache.ignite.IgniteException;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Wrapped striped executor which delegates all jobs to target executor service.
+ * It is used instead of the striped executor when that one is disabled (striped pool size <= 0).
+ */
+public class StripedExecutorProxy extends StripedExecutor {
+    /** Target executor service */
+    private volatile ExecutorService executor;
```

Comment:
It is not final because the original constructor wasn't overridden here: StripedExecutorProxy(int cnt, String igniteInstanceName, String poolName, final IgniteLogger log, boolean stealTasks)

I can override it and mark the field as final.